DynamicSizePool.hpp

```diff
@@ -94,11 +94,10 @@ class DynamicSizePool
           << numFreeBlocks() << " Free Blocks, "
           << numUsedBlocks() << " Used Blocks\n"
       );
-      std::size_t bytesFreed = freeReleasedBlocks();
+      freeReleasedBlocks();
       UMPIRE_LOG(Error, 
           "\n\tMemory exhausted at allocation resource.  "
-          "Able to give " << bytesFreed << " bytes back to recource\n"
```

Comment:
Do you think this message was useful? If so, we could just use the [`UMPIRE_USE_VAR`](https://github.com/LLNL/Umpire/blob/develop/src/umpire/util/Macros.hpp#L99) macro to silence the warning.

Reply:
It turns out that this information can be derived from the rest of the log messages (previously logged allocatedSize() - currently logged allocatedSize())

Reply:
In that case, merge away!